pandas/core/indexes/base.py

```diff
@@ -4938,6 +4938,26 @@ def get_indexer_for(self, target, **kwargs):
         indexer, _ = self.get_indexer_non_unique(target)
         return indexer
 
+    def _get_indexer_non_comparable(self, target, method, unique: bool = True):
+        """
+        If we have non-comparable dtypes, there will never be any matches.
+
+        For get_indexer calls with a method, this is an _inequality_ check
+        which raises TypeError.
+        """
```

Comment:
can you add a doc-string of Parametes / Returns / Raises. I think part of the confusion in indexing is that our doc-strings are not very clear on these things (e.g. what is supposed to raise etc).

Reply:
updated + greenish